src/libcollections/string.rs

```diff
@@ -1015,7 +1015,7 @@ impl AsRef<str> for String {
 impl<'a> From<&'a str> for String {
     #[inline]
     fn from(s: &'a str) -> String {
-        s.to_string()
+        String { vec: <[_]>::to_vec(string.as_bytes()) }
```

Comment:
As an onlooker, I'm a bit confused here. Where is the `string` variable coming from, or do you mean to have `s`?